### PR TITLE
Biomass technologies and carbon flow calculations

### PR DESCRIPTION
**Pull request by millingermarkus**

### CODE REVIEW

File: scripts/compile_cost_assumptions.py [hunk 1]
```diff
@@ -1247,9 +1453,11 @@ def get_factor(df, cols, utility_col):
 d_by_tech = get_data_from_DEA(data_in, expectation=snakemake.config["expectation"])
 # concat into pd.Dataframe
 tech_data = pd.concat(d_by_tech).sort_index()
+
+print('Pre-cleaning: ', tech_data.head(50))
 # clean up units
 tech_data = clean_up_units(tech_data)
-
+print(tech_data.head(50))
```
Comment (fneum) [inline scripts/compile_cost_assumptions.py:1460]: delete

File: scripts/compile_cost_assumptions.py [hunk 2]
```diff
@@ -1247,9 +1453,11 @@ def get_factor(df, cols, utility_col):
 d_by_tech = get_data_from_DEA(data_in, expectation=snakemake.config["expectation"])
 # concat into pd.Dataframe
 tech_data = pd.concat(d_by_tech).sort_index()
+
+print('Pre-cleaning: ', tech_data.head(50))
```
Comment (fneum) [inline scripts/compile_cost_assumptions.py:1457]:
```suggestion
print('Pre-cleaning: ', tech_data.head())
```
maybe a few less

File: config.yaml [hunk 3]
```diff
@@ -26,4 +26,4 @@ offwind_no_gridcosts : true
 desalination:
     salinity: 35 # in PSU (Practical Salinity Unit) = kg/m^3
 
-ndigits: 2
\ No newline at end of file
+ndigits: 5
```
[the remark this answers is not in the document]
Comment (millingermarkus) [reply config.yaml:29]: Not anymore. It was needed for the carbon balances but if the negative emissions from biomass are substracted at the point of combustion this should not be an issue anymore.

File: scripts/compile_cost_assumptions.py [hunk 4]
```diff
@@ -65,7 +66,9 @@
                'biomass HOP': '09c Straw HOP',
                'central coal CHP': '01 Coal CHP',
                'central gas CHP': '04 Gas turb. simple cycle, L',
-               'central solid biomass CHP': '09b Wood Pellets, Medium',
+               'central solid biomass CHP': '09a Wood Chips, Large 50 degree',#'09b Wood Pellets, Medium',
```
[the remark this answers is not in the document]
Comment (millingermarkus) [reply scripts/compile_cost_assumptions.py:69]: Biomass CHP generally use wood chips and not pellets which are more expensive.

File: scripts/compile_cost_assumptions.py [hunk 5]
```diff
@@ -115,6 +126,7 @@
                     'central coal CHP': '',
                     'central gas CHP': 'I:J',
                     'central solid biomass CHP': 'I:J',
+                    # 'central solid biomass CHP CCS': 'I:J',
```
[the remark this answers is not in the document]
Comment (millingermarkus) [reply scripts/compile_cost_assumptions.py:129]: redundant for now

File: scripts/compile_cost_assumptions.py [hunk 6]
```diff
@@ -387,10 +416,36 @@ def add_conventional_data(costs):
...
+    costs.loc[('nuclear_new', 'efficiency'), 'unit'] = "per unit"
+    costs.loc[('nuclear_new', 'efficiency'), 'source'] = source_dict['Lazards']
+
+    costs.loc[('nuclear_new', 'fuel'), 'value'] = 2.6
+    costs.loc[('nuclear_new', 'fuel'), 'unit'] = 'EUR/MWh_th'
+    costs.loc[('nuclear_new', 'fuel'), 'source'] = source_dict['Lazards']
+
+    costs.loc[('nuclear_new', 'lifetime'), 'value'] = 60
+    costs.loc[('nuclear_new', 'lifetime'), 'unit'] = "years"
+    costs.loc[('nuclear_new', 'lifetime'), 'source'] = "Assumption"
```
[the remark this answers is not in the document]
Comment (millingermarkus) [reply scripts/compile_cost_assumptions.py:446]: Agree

**Pull request merged**